### PR TITLE
feat: added ifft2

### LINKED ISSUE #23585: ifft2

### PR DESCRIPTION
**Pull request by roudik**
# PR Description

## Related Issue

Close #23585

## Checklist

- [ ] Did you add a function?
- [ ] Did you add the tests?
- [ ] Did you follow the steps we provided?

### CODE REVIEW

**Comment by roudik** [comment]
@hello-fri-end Hey! Please, note that the function implemented is dependent on a backend function that might not have been implemented yet. The code for ifft2 in Jax frontend should be correct, and any errors should be resulting from calling ivy.ifft2.

File: ivy_tests/test_ivy/test_frontends/test_jax/test_numpy/test_fft.py [hunk 1]
```diff
@@ -163,3 +163,54 @@ def test_jax_numpy_ifft(
         atol=1e-02,
         rtol=1e-02,
     )
+
+
+# ifft2
+@handle_frontend_test(
+    fn_tree="jax.numpy.fft.ifft2",
+    dtype_values=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("complex"),
```
Comment (hello-fri-end) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_numpy/test_fft.py:172]:
```suggestion
        available_dtypes=helpers.get_dtypes("valid"),
```

This should always be `valid` - if only `complex` data-type is supported by the function then that's what `valid` will return.

**Review by hello-fri-end — CHANGES_REQUESTED** [review]
Hey @roudik ! Thanks for working on this task, this is already a great PR. Just left a minor comments, please resolve it and I'll be happy to review the PR again. Thanks :)

File: ivy_tests/test_ivy/test_frontends/test_jax/test_numpy/test_fft.py (hunk 1 above)
Comment (roudik) [reply ivy_tests/test_ivy/test_frontends/test_jax/test_numpy/test_fft.py]: > This should always be `valid` - if only `complex` data-type is supported by the function then that's what `valid` will return.

Dear @hello-fri-end, just changed it to valid. Any other comments?

**Review by hello-fri-end — APPROVED** [review]
LGTM! Thanks for the contribution @roudik :rocket:

**Pull request merged**